backend/src/user/user.service.ts

```diff
@@ -1,44 +1,101 @@
-import { Injectable, NotFoundException } from '@nestjs/common'
+import {
+  ForbiddenException,
+  Injectable,
+  NotFoundException,
+} from '@nestjs/common'
 import { CreateUserDto, UpdateUserDto } from './dto/user.dto'
-
 import { UserRepository } from './user.repository'
+import { hash } from 'bcrypt'
+import { Role, User } from '@prisma/client'
 
 @Injectable()
 export class UserService {
   constructor(private readonly userRepository: UserRepository) {}
 
   async findAll() {
-    return {
-      data: await this.userRepository.findAll(),
-    }
+    return { data: await this.userRepository.findAll() }
   }
 
-  async findOne(id: number) {
-    const user = await this.userRepository.findOne(id)
+  async findByEmail(email: string) {
+    const user = await this.userRepository.findByEmail(email)
 
     if (!user) {
-      throw new NotFoundException('User with given ID not found')
+      throw new NotFoundException('User with given email not found')
     }
 
     return {
       data: user,
     }
   }
 
-  async create(data: CreateUserDto) {
+  async findById(id: number, user?: User) {
+    const foundUser = await this.userRepository.findById(id)
+    if (user) user = (await this.findByEmail(user.email)).data
```

Comment:
Nie mam pewności czy to ma sens :) albo `if(!user) user = ... ` albo w ogóle trzeba to wywalić, bo może currentUser zawsze jest.